ANMN/LTSP/TSaggregator/geoserverCatalog_generic.py

```diff
@@ -0,0 +1,112 @@
+# -*- coding: utf-8 -*-
+"""
+geoserverCatalog.py
+
+Collect files names from the AODN geoserver according to several conditions
+Output a list of urls and optionally write into a text file
+
+"""
+
+
+import sys
+import argparse
+from datetime import datetime, timedelta
+from distutils.util import strtobool
+
+import pandas as pd
+
+
+
+def args():
+    parser = argparse.ArgumentParser(description="Get a list of urls from the AODN geoserver")
+    parser.add_argument('-var', dest='varname', help='name of the variable of interest, like TEMP', default=None, required=False)
+    parser.add_argument('-site', dest='site', help='site code, like NRMMAI',  type=str, default=None, required=False)
+    parser.add_argument('-ft', dest='featuretype', help='feature type, like timeseries', default=None, required=False)
+    parser.add_argument('-fv', dest='fileversion', help='file version, like 1', default=None, type=int, required=False)
+    parser.add_argument('-ts', dest='timestart', help='start time like 2015-12-01', default=None, type=str, required=False)
+    parser.add_argument('-te', dest='timeend', help='end time like 2018-06-30', type=str, default=None, required=False)
+    parser.add_argument('-out', dest='outFileList', help='name of the file to store the selected files urls', default=None, required=False)
+    parser.add_argument('-realtime', dest='realtime', help='yes or no. If absent, all modes will be retrieved', type=str, default=None, required=False)
+
+    vargs = parser.parse_args()
+    return(vargs)
+
+
+def get_urls(varname=None, site=None, featuretype=None, fileversion=None, realtime=None, timestart=None, timeend=None, outFileList=None):
+
+    """
+    get the urls from the geoserver moorings_all_map collection
+    based on user defined filters
+    """
+
+    if realtime:
+        if realtime.lower() == "yes":
+            url = "http://geoserver-123.aodn.org.au/geoserver/ows?typeName=moorings_all_map&SERVICE=WFS&REQUEST=GetFeature&VERSION=1.0.0&outputFormat=csv&CQL_FILTER=(realtime=TRUE)"
+        elif realtime.lower() == "no":
+            url = "http://geoserver-123.aodn.org.au/geoserver/ows?typeName=moorings_all_map&SERVICE=WFS&REQUEST=GetFeature&VERSION=1.0.0&outputFormat=csv&CQL_FILTER=(realtime=FALSE)"
+        else:
+            sys.exit('ERROR: %s is not yes or no' % realtime)
+    else:
+        url = "http://geoserver-123.aodn.org.au/geoserver/ows?typeName=moorings_all_map&SERVICE=WFS&REQUEST=GetFeature&VERSION=1.0.0&outputFormat=csv"
+
+
+    df = pd.read_csv(url)
+    criteria_all = df.url != None
+
+    ## get possible values
+    separator = ', '
+    varnames_all = set(separator.join(list(df.variables)).split(', '))
+    varnames_all = set(str(list(df.variables)).split(", "))
+    site_all = list(df.site_code.str.upper().unique())
+    featuretype_all = list(df.feature_type. str.lower().unique())
+    fileversion_all = list(df.file_version.unique())
+    mode_all = ['realtime', 'delayed', 'all']
```

Comment:
Same  with all  of  these `_all` variables. No point working them out if they're not used. Also,
* `site_code` is case sensitive, so don't use `.upper()` for `site_all`
* `feature_type.lower()` should be "timeseries", "profile", or "timeseriesprofile". You don't need to work that out from 46k rows every time.
* `file_version` will be 0, 1, or 2. 
* `mode_all` is not used anywhere, so can be removed.

Reply:
fixed